pkg/upgrade/hub/hub.go

```diff
@@ -94,18 +87,21 @@ func (h *HubClient) MountURL(flist string) string {
 
 // StorageURL return hub storage url
 func (h *HubClient) StorageURL() string {
-	if kernel.GetParams().IsV4() {
-		return hubv4Storage
-	}
-	return hubStorage
+	env := environment.MustGet()
+	return env.HubStorage
```

Comment:
this should return the hubstorage based on if this is v3 or v4 here u r returning one hubstorage all the time
```
hubStorage   = "zdb://hub.threefold.me:9900"
	hubv4Storage = "zdb://hub.threefold.me:9940"
	``` 
they have diff port

Reply:
environment.Must get returns hubstorage with v3 if the node is of type v3 and v4 if the node is of type v4, so we don't need to do the check again here

check: https://github.com/threefoldtech/zosbase/pull/33/files#diff-616422b9b983db329c853b3e73274dfdb95b49add7e27bcfd47257923eead1e8R391-R401